src/tools/compiletest/src/main.rs

```diff
@@ -484,6 +484,11 @@ pub fn run_tests(config: &Config) {
 
     // FIXME(#33435) Avoid spurious failures in codegen-units/partitioning tests.
     if let Mode::CodegenUnits = config.mode {
+        // codegen-units tests use wrong directory, for more info see
+        // https://github.com/rust-lang/rust/issues/34586#issuecomment-501681886
+        if let Ok(build_dir) = env::var("BUILD_DIR") {
```

Comment:
Hmm, if the problem is that we are passing `-Zincremental=tmp/foo`, maybe it would be better to have some way to alter the command-line flags to include the correct directory?

I'm imagining:

```
// compile-flags:-Zincremental=$BUILD_DIR/tmp/...
```

for example? (We'd have to replace that `$BUILD_DIR` of course.)

Reply:
I'm a bit worried about side-effects from changing the current directory in a narrow range of cases.

Reply:
Environment variables doesn't seem to be expanded in `compile-flags` and that results in `$VARIABLE/tmp/...`.

> I'm a bit worried about side-effects from changing the current directory in a narrow range of cases.

When I enabled it for everything only 3 rustdoc tests broke(I was building with `--no-fail-fast`), I haven't investigated further but they should be fixable.